flytekit/image_spec/image_spec.py

```diff
@@ -107,6 +107,8 @@ def is_container(self) -> bool:
     def exist(self) -> bool:
         """
         Check if the image exists in the registry.
+        Return True if the image exists in the registry, False otherwise.
+        Return None if failed to check if the image exists due to the permission issue or other reasons.
```

Comment:
That Python typing for `exist` needs to be updated to include `None`.

Reply:
updated